nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -400,29 +400,29 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
     if (!withinAnnotatedCode(state)) {
       return Description.NO_MATCH;
     }
-    Symbol.MethodSymbol methodSymbol = getSymbolForMethodInvocation(tree, state);
+    Symbol.MethodSymbol methodSymbol = getSymbolForMethodInvocation(tree);
     handler.onMatchMethodInvocation(this, tree, state, methodSymbol);
     // assuming this list does not include the receiver
     List<? extends ExpressionTree> actualParams = tree.getArguments();
     return handleInvocation(tree, state, methodSymbol, actualParams);
   }
 
-  private static Symbol.MethodSymbol getSymbolForMethodInvocation(
-      MethodInvocationTree tree, VisitorState state) {
+  private static Symbol.MethodSymbol getSymbolForMethodInvocation(MethodInvocationTree tree) {
     Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
     Verify.verify(methodSymbol != null, "not expecting unresolved method here");
+    // In certain cases, we need to get the base symbol for the method rather than the symbol
+    // attached to the call.
     // For interface methods, if the method is an implicit method corresponding to a method from
-    // java.lang.Object, use the symbol for the java.lang.Object method instead.  We do this to
+    // java.lang.Object, the base symbol is for the java.lang.Object method.  We need this to
     // properly treat the method as unannotated, which is particularly important for equals()
     // methods.  This is an adaptation to a change in JDK 18; see
     // https://bugs.openjdk.org/browse/JDK-8272564
-    if (methodSymbol.owner.isInterface()) {
-      Symbol.MethodSymbol baseSymbol = (Symbol.MethodSymbol) methodSymbol.baseSymbol();
-      if (baseSymbol != methodSymbol && baseSymbol.owner == state.getSymtab().objectType.tsym) {
-        methodSymbol = baseSymbol;
-      }
-    }
-    return methodSymbol;
```

Comment:
question: I'm a bit confused about the previous code. It suggests that we actually want the `methodSymbol` and we only resolve to `baseSymbol` under certain conditions (i.e., interface method corresponding to one from `java.lang.Object`). But it seems with the updated code that we actually always want `baseSymbol`? 

Am I missing something?

Reply:
Yes, that's a good question, this was due to my misunderstanding before.  Basically, always getting the base symbol is safe, since if there is no different base symbol the `baseSymbol()` method just returns `this`.  I didn't understand this before; more detail in [this comment](https://github.com/uber/NullAway/issues/764#issuecomment-1921863902).  In the next EP version the `getSymbol` overload for `MethodInvocationTree` will always call `baseSymbol()`, so in some cases it may get called twice, but that is fine.